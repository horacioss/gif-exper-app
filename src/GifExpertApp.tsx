import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

interface Props {
  defaultCategories?: string[];
}

const GifExpertApp: React.FC<Props> = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState<string[]>(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      {categories.map((item) => (
        <GifGrid key={item} category={item} />
      ))}
    </>
  );
};

export default GifExpertApp;
