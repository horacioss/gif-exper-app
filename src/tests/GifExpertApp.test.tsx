import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("tests over <GifExpertApp />", () => {
  test("should render <GifExpertApp /> corectly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should have 2 <GifGrid /> elements", () => {
    const categories = ["Samurai X", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
