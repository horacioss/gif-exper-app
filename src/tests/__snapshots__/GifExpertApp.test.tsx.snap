// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests over <GifExpertApp /> should have 2 <GifGrid /> elements 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Samurai X"
    key="Samurai X"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Fragment>
`;

exports[`tests over <GifExpertApp /> should render <GifExpertApp /> corectly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
